cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/DefaultZooKeeperAdminProvider.java

```diff
@@ -20,12 +20,30 @@ public class DefaultZooKeeperAdminProvider implements ZooKeeperAdminProvider {
      * @param connectString     Connection String used to connect to Zookeeper
      * @param sessionTimeout    Session timeout
      * @param watcher           Watcher which will be notified about watches and connection changes
-     * @param conf              Zookeeper client configuration
+     * @param operationTimeoutMs    Timeout for ZooKeeper requests
+     * @param trustStoreFile        File hosting the truststore with TLS certificates to use to connect to ZooKeeper
+     * @param keyStoreFile          File hosting the keystore with TLS private keys to use to connect to ZooKeeper
      *
      * @return  ZooKeeperAdmin instance
      */
     @Override
-    public ZooKeeperAdmin createZookeeperAdmin(String connectString, int sessionTimeout, Watcher watcher, ZKClientConfig conf) throws IOException {
-        return new ZooKeeperAdmin(connectString, sessionTimeout, watcher, conf);
+    public ZooKeeperAdmin createZookeeperAdmin(String connectString, int sessionTimeout, Watcher watcher,
+                                               long operationTimeoutMs, String trustStoreFile, String keyStoreFile) throws IOException {
+        ZKClientConfig clientConfig;
+        try {
+            clientConfig = new ZKClientConfig();
+        } catch (Exception e)    {
```

Comment:
What exception do you expect it to throw?

Reply:
The previous version of this code (in ZooKeeperScaler) just caught the generic class Exception. When I was doing the refactor I checked and thought it might throw a SecurityException if the call to System.getProperty fails in the ZKClientConfig class. However, on further reading I think actually that's unlikely as it requires a SecurityManager to be configured. So I would propose we don't do a try catch. What do you think @scholzj?